packages/chips/Chip.js

```diff
@@ -21,26 +21,39 @@ export class Chip extends Component {
 
   get classes() {
     const {classList} = this.state;
-    const {className} = this.props;
-    return classnames('mdc-chip', Array.from(classList), className);
+    const {className, selected} = this.props;
+    return classnames('mdc-chip', Array.from(classList), className, {
+      'mdc-chip--selected': selected,
+    });
   }
 
   get adapter() {
     return {
-      addClass: (className) =>
-        this.setState({classList: this.state.classList.add(className)}),
+      addClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
+      },
       removeClass: (className) => {
-        const {classList} = this.state;
+        const classList = new Set(this.state.classList);
         classList.delete(className);
         this.setState({classList});
       },
+      hasClass: (className) => this.classes.split(' ').includes(className),
     };
   }
 
+  handleInteraction = () => {
+    this.props.handleSelect(this.props.id);
```

Comment:
Its possible that the users of this chips component will want to do something onClick. Also I think changing from `handleInteraction` --> `handleClick` is more clear. Try:

```
handleClick = (e) => {
  if (typeof this.props.onClick === 'function) {
    this.props.onClick(e);
  }
  this.props.handleSelect(this.props.id);
}
```